x-pack/legacy/plugins/monitoring/public/views/base_controller.js

```diff
@@ -110,6 +128,22 @@ export class MonitoringViewBaseController {
       timefilter.enableAutoRefreshSelector();
     }
 
+    const updateHandler = () => {
+      const { from, to } = timefilter.getTime();
+      const parsedRange = {
+        from: dateMath.parse(from),
+        to: dateMath.parse(to),
+      };
+
+      if (invalidTimeRange(parsedRange)) {
+        timefilter.setTime(VALID_RANGE);
```

Comment:
This will always reset the timepicker range to `Last 30 seconds` if an invalid time range is used. I'm not sure we should do that.

Consider the scenario where a user is viewing data from `8 minutes ago` to `6 minutes ago`. Then they try to change the time picker upper bound to `8 minutes ago`. They will get the invalid time range toast (which is good) but then their time picker will get reset to `Last 30 seconds`, which might be a bit surprising to the user since.

I think a slightly better UX might be to show them the toast but leave their timepicker bounds unchanged (so that would be `8 minutes ago` to `6 minutes ago` in my example above).

WDYT?

Reply:
++ Agreed

Reply:
++ Good point 👍